Braze Demo/AppboyManager.swift

```diff
@@ -237,3 +259,37 @@ extension Notification.Name {
   static let homeScreenContentCard = Notification.Name("kHomeScreenContentCard")
   static let reorderHomeScreen = Notification.Name("kReorderHomeScreen")
 }
+
+// MARK: - Modal In App Message
+class ModalViewController: ABKInAppMessageModalViewController {
+  
+  // MARK: - Outlets
+  @IBOutlet private weak var primaryButton: ABKInAppMessageUIButton!
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    if let immersiveMessage = inAppMessage as? ABKInAppMessageImmersive, let buttons = immersiveMessage.buttons {
+      switch buttons.count {
+      case 1:
+        primaryButton.titleLabel?.text = buttons[0].buttonText
+      case 2:
+        primaryButton.titleLabel?.text = buttons[1].buttonText
+      default:
+        break
+      }
+    }
+  }
+}
+
+// MARK: - In-App Message View Controller Helper
+private extension AppboyManager {
+  func modalViewController(inAppMessage: ABKInAppMessage) -> ABKInAppMessageModalViewController {
+    switch inAppMessage.extras?["view_type"] as? String {
```

Comment:
Consider constantizing "view_type". That would also help with keeping the comment reference to "view_type " in `InAppMessageData.swift` up to date